### ISSUE TITLE
@storybook/addon-centered/angular does not pass defined props to component

### ISSUE THREAD

**Issue by MaximSagan** [storybookjs/storybook#6748]
**Describe the bug**

**To Reproduce**
1. Create a component that requires props
2. Add the 'centered' docrator
3. Specify your component using the `component` property.
4. Notice that defined props are not observed by the rendered component. Internally, the component believes they areall  `undefined`.

Example code:
```ts
storiesOf(`${MyComponent.name} stories`, module)
  .addDecorator(centered)
  .add(`${MyComponent.name} story`, () => ({
    component: MyComponent,
    props: {
      mainText: 'I like shorts because they\'re fun and comfy to wear.',
    }
}));
```

**Expected behavior**
Props should be passed to specified component

**Comment by eddiegroves** [comment]
This is still happening in the latest version - props are ignored.

**Comment by shilman** [comment]
@eddiegroves 5.2?

**Comment by eddiegroves** [comment]
> @eddiegroves 5.2?

Yep 5.2

**Comment by gaetanmaisse** [comment]
Hey @eddiegroves, I wasn't able to reproduce it. I tried on a side project and on my company Angular monorepo and for both `props` are working.  Can you provide a simple repository that reproduces this bug?

**Comment by ramarivera** [comment]
Hi @gaetanmaisse , I also came across this issue today. The problem seems to show up only when using the component syntax (as described here https://github.com/storybookjs/storybook/issues/3836#issuecomment-403743428). 

I tracked the problem and it seems that while the centered decorator gets the component and props as defined in the story (https://github.com/storybookjs/storybook/blob/master/addons/centered/src/angular.ts#L50):

```
function centered(metadataFn: StoryFn<IStory>) {
  const metadata = metadataFn();

  return {
    ...metadata,
    template: getTemplate(metadata),
    moduleMetadata: getModuleMetadata(metadata),
    props: {
      ...metadata.props,
      styles,
    },
  };
}
```
when processing the component in order to get the template, if the component syntax is used the props are not set in the created template (https://github.com/storybookjs/storybook/blob/master/addons/centered/src/angular.ts#L13)
```
function getTemplate(metadata: any) {
  let tpl = '';
  if (metadata.component) {
    const selector = getComponentSelector(metadata.component);
    tpl = `<${selector}></${selector}>`;
  }

  if (metadata.template) {
    tpl = metadata.template;
  }

  return `      
      <div [ngStyle]="styles.style">
        <div [ngStyle]="styles.innerStyle">
          ${tpl}
        </div>
      </div>`;
}
```

The template syntax works because the centered component does have access to the props and since the names the template expect the props to have matches the actual props defined in the story, it just works.

I think this also addresses these two issues: #4021 #3836

I think I can fix it with a PR, unless someone else is on it already? Let me know what you think.

And btw I started using Storybook yesterday and I'm completely in love!
